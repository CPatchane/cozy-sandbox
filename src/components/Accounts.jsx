import React from 'react'

import { Button } from 'cozy-ui/react/Button'
import IntentOpener from 'cozy-ui/react/IntentOpener'

const Accounts = () => (
  <div>
    <h1>Accounts</h1>
    <IntentOpener
      action="CREATE"
      doctype="io.cozy.accounts"
      onComplete={res => alert('intent has completed ! ' + JSON.stringify(res))}
      onDismiss={() => alert('intent has been dismissed !')}
      options={{
        slug: 'bouyguesbox-collect'
      }}
    >
      <Button>Create account for Bouygues Box</Button>
    </IntentOpener>
  </div>
)

export default Accounts
