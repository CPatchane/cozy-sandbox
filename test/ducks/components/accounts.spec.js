'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Accounts from '../../../src/components/Accounts'

Enzyme.configure({ adapter: new Adapter() })

describe('Accounts', () => {
  it('should be rendered correctly', () => {
    const component = shallow(<Accounts />).getElement()
    expect(component).toMatchSnapshot()
  })
})
