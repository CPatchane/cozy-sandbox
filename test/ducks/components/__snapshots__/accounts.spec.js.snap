// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts should be rendered correctly 1`] = `
<div>
  <h1>
    Accounts
  </h1>
  <IntentOpener
    action="CREATE"
    closable={true}
    doctype="io.cozy.accounts"
    onComplete={[Function]}
    onDismiss={[Function]}
    options={
      Object {
        "slug": "trainline",
      }
    }
    tag="span"
  >
    <Button
      busy={false}
      className=""
      disabled={false}
      icon={null}
      label={null}
      onClick={null}
      theme=""
      type="submit"
    >
      Create account for Bouygues Box
    </Button>
  </IntentOpener>
</div>
`;
